HoneycombManager.cs

```diff
@@ -0,0 +1,263 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+using UnityEngine.Tilemaps;
+using UnityEditor;
+
+public class HoneycombManager : MonoBehaviour
+{
+    [SerializeField] Tilemap hiveGrid;
+    [SerializeField] Tilemap buyGrid;
+    [SerializeField] Tilemap BGGrid;
+    [SerializeField] Tilemap BGGridTemp;
+    [SerializeField] Tile[] hcTile;
+    [SerializeField] Tile[] hcNectarTile;
+    [SerializeField] Tile buyTile;
+    [SerializeField] Tile buyTileEmpty;
+    [SerializeField] GameObject HCPriceObj;
+    [SerializeField] GameObject HCPriceTxtObj;
+    [SerializeField] GameObject StorageManager;
+    [SerializeField] GameObject InputManager;
+    [SerializeField] GameObject BGManager;
+    [SerializeField] GameObject CameraManager;
+
+    BuyHoneycombText HCPriceTxt;
+    StorageManager SMScript;
+    InputManager IPScript;
+    HiveBGManager BGScript;
+    CameraManager CMScript;
+
+    private Vector3Int mouseTilePos;
+
+    Camera camera;
+
+    private List<int[]> hcPos;
+    private List<float> honeyStorage;
+    private List<float> nectarStorage;
+    private List<int> honeyStorageM;
+    private List<int> nectarStorageM;
+
+    private int honeycombNum = 0;
+    private float HCPrice = 600;
+    private int HCPriceM = 0;
+
+    private int left;
+    private int right;
+    private int up;
+    private int down;
+
+    private float honey;
+    private float nectar;
+    private float pollen;
+    private float wax;
+
+    private int honeyM;
+    private int nectarM;
+    private int pollenM;
+    private int waxM;
+
+    private float honeyCapacity = 150;
+    private int honeyCapacityM = 0;
+    private float nectarCapacity = 150;
+    private int nectarCapacityM = 0;
+
+    private string HCPriceStr;
+
+    public int getLeft(){ return left; }
+    public int getRight(){ return right; }
+    public int getUp(){ return up; }
+    public int getDown(){ return down; }
+    public int getHoneycombNum(){ return honeycombNum; }
```

Comment:
C#에는 property라는 문법이 있습니다.
Get, Set으로 자주 접근되는 변수를 함수 호출이 아닌 변수와 비슷한 형태로 호출하는 건데요.
[속성 - C# 공식문서](https://docs.microsoft.com/ko-kr/dotnet/csharp/programming-guide/classes-and-structs/properties)
매우 예쁘고 직관적이니 추천합니다. property를 이용해서 이렇게 쓸 수 있겠죠
```suggestion
    public int Left { get; }
    public int Right { get; }
    public int Up { get; }
    public int Down { get; }
    public int HoneycombNum { get; }
```